src/Dotnet.Script.Core/ScriptCompiler.cs

```diff
@@ -68,11 +75,11 @@ public ScriptCompiler(ScriptLogger logger, RuntimeDependencyResolver runtimeDepe
             RuntimeDependencyResolver = runtimeDependencyResolver;
         }
 
-        public virtual ScriptOptions CreateScriptOptions(ScriptContext context)
-        {
+        private ScriptOptions CreateScriptOptions(ScriptContext context, IDictionary<string, IReadOnlyList<string>> scriptMap)
```

Comment:
I would rather keep it `public virtual` I have some ideas on how to consume this core library i.e. inside an ASP.NET Core app and would like to keep this overrideable. I'd suggest to pass entire `IList<RuntimeDependency>` to this method and pick up script map inside it, this way the public API is cleaner

Reply:
No problem. Fixing